src/main/java/org/greencodeinitiative/creedengo/java/checks/DataInHibernateMustBePaginated.java

```diff
@@ -0,0 +1,128 @@
+package org.greencodeinitiative.creedengo.java.checks;
+
+import java.util.*;
+
+import org.sonar.check.Rule;
+import org.sonar.plugins.java.api.InputFileScannerContext;
+import org.sonar.plugins.java.api.IssuableSubscriptionVisitor;
+import org.sonar.plugins.java.api.tree.*;
+import org.sonar.plugins.java.api.tree.Tree.Kind;
+import org.sonar.plugins.java.api.semantic.Symbol;
+import org.sonar.plugins.java.api.semantic.Type;
+
+
+import java.util.Collections;
+import java.util.List;
+
+import org.sonar.check.Rule;
+import org.sonar.plugins.java.api.IssuableSubscriptionVisitor;
+import org.sonar.plugins.java.api.tree.*;
+import org.sonar.plugins.java.api.semantic.Symbol;
+import org.sonar.plugins.java.api.semantic.Type;
+
+import java.util.Collections;
+import java.util.List;
+
+@Rule(key = "GCI1111")
+public class DataInHibernateMustBePaginated extends IssuableSubscriptionVisitor {
+
+    private static final String MESSAGE = "Hibernate queries must be paginated to avoid excessive data loading";
+
+    @Override
+    public List<Tree.Kind> nodesToVisit() {
+        return Collections.singletonList(Kind.METHOD);
+    }
+
+    @Override
+    public void visitNode(Tree tree) {
+        MethodTree methodTree = (MethodTree) tree;
+        ClassTree enclosingClass = getEnclosingClass(tree);
+
+        if (enclosingClass == null || !isRepository(enclosingClass)) {
+            return; // Ne rien faire si ce n’est pas un Repository
+        }
+
+
+        // Récupérer le nom de la méthode
+        String methodName = methodTree.simpleName().name();
+
+        // Récupérer le type de retour
+        String returnType = methodTree.returnType().symbolType().toString();
+
+        // Vérifier s’il retourne une collection entière (sans pagination)
+        boolean returnsAllData = retrournAllData(returnType);
+
+        // Vérifier s’il utilise la pagination
+        boolean hasPaginationParam = isHasPaginationParam(methodTree);
+
+        // Vérifier la présence de l’annotation @Query
+        boolean usesQueryAnnotation = isUsesQueryAnnotation(methodTree);
+
+        extracted(returnsAllData, hasPaginationParam, methodTree, usesQueryAnnotation);
+
+    }
+
+    private void extracted(boolean returnsAllData, boolean hasPaginationParam, MethodTree methodTree, boolean usesQueryAnnotation) {
```

Comment:
please use a method name more explicit ... when I read the method name I must understand what it will do